### PR TITLE
[vcpkg baseline][gdal] Fix configure error on OSX

### LINKED ISSUE #16990: [gdal:x64-osx] configuration failure: required file 'config.rpath' not found
**Host Environment**
- OS: osx
- Compiler: n/a

**To Reproduce**
`./vcpkg install gdal`

**Failure logs** 
stdout logs
```
Computing installation plan...
The following packages will be built and installed:
    gdal[core]:x64-osx -> 3.1.3#3
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: /Users/andrewdelpit/.cache/vcpkg/archives/2c/2c2d63f1b9da6dc2bf4c59cb0cbb2188264546a1.zip
Starting package 1/1: gdal:x64-osx
Building package gdal[core]:x64-osx...
-- Using cached /Users/andrewdelpit/repo/photogrammetry/vcpkg/downloads/gdal313.zip
-- Cleaning sources at /Users/andrewdelpit/repo/photogrammetry/vcpkg/buildtrees/gdal/src/gdal313-b1dafdc415.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/andrewdelpit/repo/photogrammetry/vcpkg/downloads/gdal313.zip
-- Applying patch 0001-Fix-debug-crt-flags.patch
-- Applying patch 0002-Fix-build.patch
-- Applying patch 0003-Fix-static-build.patch
-- Applying patch 0004-Fix-std-fabs.patch
-- Applying patch 0005-Fix-cfitsio.patch
-- Using source at /Users/andrewdelpit/repo/photogrammetry/vcpkg/buildtrees/gdal/src/gdal313-b1dafdc415.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Generating configure for x64-osx
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: /usr/local/bin/autoreconf -vfi
    Working Directory: /Users/andrewdelpit/repo/photogrammetry/vcpkg/buildtrees/gdal/src/gdal313-b1dafdc415.clean/
    Error code: 1
    See logs for more information:
      /Users/andrewdelpit/repo/photogrammetry/vcpkg/buildtrees/gdal/autoconf-x64-osx-out.log
      /Users/andrewdelpit/repo/photogrammetry/vcpkg/buildtrees/gdal/autoconf-x64-osx-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:578 (vcpkg_execute_required_process)
  ports/gdal/portfile.cmake:191 (vcpkg_configure_make)
  scripts/ports.cmake:142 (include)


Error: Building package gdal:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gdal:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
```

And the contents of `autoconf-x64-osx-err.log`
```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: running: glibtoolize --copy --force
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force 
autoreconf: running: /usr/local/Cellar/autoconf/2.71/bin/autoconf --force
configure.ac:49: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:49: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/ax_cflags_warn_all.m4:26: AX_CFLAGS_WARN_ALL is expanded from...
configure.ac:49: the top level
configure.ac:50: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:50: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/ax_cflags_warn_all.m4:71: AX_CXXFLAGS_WARN_ALL is expanded from...
configure.ac:50: the top level
configure.ac:222: warning: The macro `AC_LIBTOOL_WIN32_DLL' is obsolete.
configure.ac:222: You should run autoupdate.
m4/ltoptions.m4:148: AC_LIBTOOL_WIN32_DLL is expanded from...
configure.ac:222: the top level
configure.ac:222: warning: AC_LIBTOOL_WIN32_DLL: Remove this warning and the call to _LT_SET_OPTION when you
configure.ac:222: put the 'win32-dll' option into LT_INIT's first parameter.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
m4/ltoptions.m4:148: AC_LIBTOOL_WIN32_DLL is expanded from...
configure.ac:222: the top level
configure.ac:223: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:223: You should run autoupdate.
m4/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:223: the top level
configure.ac:297: warning: The macro `AC_HEADER_STDC' is obsolete.
configure.ac:297: You should run autoupdate.
./lib/autoconf/headers.m4:704: AC_HEADER_STDC is expanded from...
configure.ac:297: the top level
configure.ac:350: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:350: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:350: the top level
configure.ac:376: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:376: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:376: the top level
configure.ac:505: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:505: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:505: the top level
configure.ac:980: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:980: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:980: the top level
configure.ac:999: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:999: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:999: the top level
configure.ac:1016: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:1016: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:1016: the top level
configure.ac:1037: warning: The macro `AC_CHECKING' is obsolete.
configure.ac:1037: You should run autoupdate.
./lib/autoconf/general.m4:2499: AC_CHECKING is expanded from...
configure.ac:1037: the top level
configure.ac:1041: warning: The macro `AC_CHECKING' is obsolete.
configure.ac:1041: You should run autoupdate.
./lib/autoconf/general.m4:2499: AC_CHECKING is expanded from...
configure.ac:1041: the top level
configure.ac:1058: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:1058: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:1058: the top level
configure.ac:1079: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:1079: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:1079: the top level
configure.ac:3052: warning: AC_CHECK_HEADERS($NETCDF_INCLUDEDIR/netcdf_mem.h): you should use literals
./lib/autoconf/headers.m4:217: AC_CHECK_HEADERS is expanded from...
configure.ac:3052: the top level
configure.ac:3672: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:3672: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_oracle_oci.m4:41: AX_LIB_ORACLE_OCI is expanded from...
configure.ac:3672: the top level
configure.ac:3792: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:3792: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_lib_xerces.m4:37: AX_LIB_XERCES is expanded from...
configure.ac:3792: the top level
configure.ac:3816: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:3816: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_lib_expat.m4:37: AX_LIB_EXPAT is expanded from...
configure.ac:3816: the top level
configure.ac:3839: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:3839: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_lib_libkml.m4:48: AX_LIB_LIBKML is expanded from...
configure.ac:3839: the top level
configure.ac:3928: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:3928: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_oracle_oci.m4:41: AX_LIB_ORACLE_OCI is expanded from...
configure.ac:3928: the top level
configure.ac:4204: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:4204: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_lib_sqlite3.m4:32: AX_LIB_SQLITE3 is expanded from...
configure.ac:4204: the top level
configure.ac:4239: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:4239: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
m4/ax_lib_sqlite3.m4:32: AX_LIB_SQLITE3 is expanded from...
configure.ac:4239: the top level
configure.ac:4448: warning: The macro `AC_CHECKING' is obsolete.
configure.ac:4448: You should run autoupdate.
./lib/autoconf/general.m4:2499: AC_CHECKING is expanded from...
configure.ac:4448: the top level
configure.ac:5611: warning: The macro `AC_CHECKING' is obsolete.
configure.ac:5611: You should run autoupdate.
./lib/autoconf/general.m4:2499: AC_CHECKING is expanded from...
configure.ac:5611: the top level
configure.ac:6009: warning: AC_OUTPUT should be used without arguments.
configure.ac:6009: You should run autoupdate.
autoreconf: running: /usr/local/Cellar/autoconf/2.71/bin/autoheader --force
autoreconf: configure.ac: not using Automake
autoreconf: './config.sub' is updated
autoreconf: './config.guess' is updated
configure.ac: error: required file 'config.rpath' not found
autoreconf: Leaving directory '.'
```

**Additional context**

If I add `execute_process(COMMAND touch ${SOURCE_PATH}/config.rpath)` immediately before `vcpkg_configure_make` then the configuration completes fine

### PR DESCRIPTION
**Pull request by JackBoosY**
Fixes #16990.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@cenit @Neumann-A do you have any suggestions about this?

Thanks.

File: ports/gdal/portfile.cmake [hunk 1]
```diff
@@ -191,6 +191,7 @@ else()
     vcpkg_configure_make(
         SOURCE_PATH ${SOURCE_PATH}
         AUTOCONFIG
+        AUTOUPDATE
```
Comment (Neumann-A) [inline ports/gdal/portfile.cmake:194]:
```suggestion
```
this is not a valid option

**Comment by Neumann-A** [comment]
https://lists.gnu.org/archive/html/bug-gettext/2011-10/msg00012.html.

Looking at the configure script it is only using AM_ICONV

**Comment by vicroms** [comment]
@Neumann-A 

You mean like the original issue poster suggested? 

```
execute_process(COMMAND touch ${SOURCE_PATH}/config.rpath)
```

**Comment by Neumann-A** [comment]
@vicroms: Yes. But actually if we assume gettext[tools] is installed we can simply copy config.rpath from there. Seems like autopoint normally would install config.rpath but the gdal configure.ac file is not calling `AM_GNU_GETTEXT([external])` before `AM_ICONV`

**Comment by JackBoosY** [comment]
Works! Thanks @Neumann-A!

**Comment by JackBoosY** [comment]
@strega-nil Ping for merge this PR quickly, this blocked many PRs.

**Review by PhoebeHui — APPROVED** [review]
LGTM, we don't have a better solution for this now, it's fine to apply this workaroud for me.

**Pull request merged**